crates/core/src/mock/fixtures/requests/starknet_getNonce.json

```diff
@@ -2,8 +2,5 @@
   "id": 1,
   "jsonrpc": "2.0",
   "method": "starknet_getNonce",
-  "params": {
-    "block_id": "latest",
-    "contract_address": "0xabde1"
-  }
+  "params": ["latest", "0xabde1"]
```

Comment:
the former param structure didn't work?
i tried it with infura on mainnet and it worked

Reply:
This is due to the implementation of the Serialize trait for `GetNonceRequestRef`, which Dojo uses in their method `send_request` when they call `let params = serde_json::to_string(&params).unwrap();`. The implementation of Serialize will serialize the data in the form of a json array, which then won't match the request if we leave it as a json object. See [here](https://github.com/xJonathanLEI/starknet-rs/blob/master/starknet-core/src/types/codegen.rs#L4396) for the implementation.